xarray/backends/api.py

```diff
@@ -398,7 +398,7 @@ def open_dataset(
     chunks: T_Chunks = None,
     cache: bool | None = None,
     decode_cf: bool | None = None,
-    mask_and_scale: bool | None = None,
+    mask_and_scale: bool | dict[str, bool] | None = None,
```

Comment:
```suggestion
    mask_and_scale: bool | Mapping[str, bool] | None = None,
```
Dicts are invariant which always causes issues...

Reply:
OK! I considered using Mapping but opted for dict as I saw it used in other places, but makes sense.